composer/core/evaluator.py

```diff
@@ -141,17 +141,27 @@ def ensure_evaluator(evaluator: Union[Evaluator, DataSpec, Iterable, Dict[str, A
         )
 
 
-def validate_eval_automicrobatching(auto_microbatching: bool, device: Device):
+def validate_evaluator(evaluator: Evaluator, device: Device):
     """Ensure automicrobatching is only on GPU.
 
     Unlike `device_train_microbatch_size`, this validation must be done separately from the
     `_is_auto_microbatching` check because `device` is not available during `Evaluator`
     initialization.
     """
+    auto_microbatching = evaluator.auto_microbatching
     if auto_microbatching and not isinstance(device, DeviceGPU):
         raise ValueError(
             'Can only use adaptive device_eval_microbatch_size on GPU. Please set device_eval_microbatch_size >= 1.',
         )
+    if evaluator.auto_microbatching and hasattr(evaluator.dataloader, 'seq_parallel_world_size'):
+        raise ValueError('Auto microbatching on evaluators is not compatible with sequence parallelism.')
+    if isinstance(evaluator.dataloader.get_num_samples_in_batch, int) and hasattr(
+        evaluator.dataloader,
+        'seq_parallel_world_size',
+    ) and evaluator.dataloader.get_num_samples_in_batch * evaluator.dataloader.seq_parallel_world_size != 1:  # type: ignore
```

Comment:
👀 what are you ignoring? Should fix

Reply:
seq_parallel_world_size is not typed into torch and is outside our control :(